index.d.ts

```diff
@@ -0,0 +1,103 @@
+/// <reference types="node" />
+
+declare module 'cloud-config-client' {
+
+    import http = require('http');
+    import https = require('https');
+
+    export function load(options: Options, callback?: LoadCallback): Promise<Config|Error>;
+
+    export abstract class Config {
+        constructor(data: ConfigData, context: { [key: string]: any });
+
+        properties(): { [key: string]: any }
+
+        raw(): { [key: string]: any }
+
+        get(keyParts: string): any
+
+        forEach(callback: (property: string, value: string) => void, includeOverridden: boolean): void
+
+        toObject(): { [key: string]: any }
+
+        toString(spaces: number): string
+    }
+
+    interface ConfigFile {
+        /** file name */
+        name: string
+
+        /** configuration properties */
+        source: ConfigSource
+    }
+
+    interface ConfigSource {}
+
+    interface ConfigData {
+        /** application name */
+        name: string
+
+        /** ist of profiles included */
+        profiles: Array<string>
+
+        /** environment label */
+        label: string
+
+        /** commit hash of properties */
+        version: string
+
+        /** properties included for application, sorted by more priority */
+        propertySources: Array<ConfigFile>
+    }
+
+    interface Spring {
```

Comment:
I don't think this interface is actually necessary. I guess it comes from the [Spring module definition](https://github.com/victorherraiz/cloud-config-client/blob/master/lib/spring.js#L4) which I think we can skip here